src/Common/src/Interop/Windows/Winsock/Interop.GetAddrInfoExW.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Net.Sockets;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+internal static partial class Interop
+{
+    internal static partial class Winsock
+    {
+        internal unsafe delegate void GetAddrInfoExCompletionCallback([In] int error, [In] int bytes, [In] NativeOverlapped* overlapped);
+
+        [DllImport(Interop.Libraries.Ws2_32, ExactSpelling = true, CharSet = CharSet.Unicode, BestFitMapping = false, ThrowOnUnmappableChar = true, SetLastError = true)]
+        internal static extern unsafe int GetAddrInfoExW(
+            [In] string name,
+            [In] string serviceName,
+            [In] int namespaceId,
+            [In] IntPtr providerId,
+            [In] ref AddressInfoEx hints,
```

Comment:
Can this use `in` instead of `ref`?

Reply:
The build is passing but Visual Studio is complaining with "Feature 'readonly references' is not available in C# 7.0. Please use language version 7.2 or greater."

I suggest leaving 'ref', I don't think it would change anything for a p/invoke call anyway.